components/callbacks/statemachine.go

```diff
@@ -96,19 +96,21 @@ func (c Callback) RegenerateTasks(*hsm.Node) ([]hsm.Task, error) {
 	case enumsspb.CALLBACK_STATE_BACKING_OFF:
 		return []hsm.Task{BackoffTask{Deadline: c.NextAttemptScheduleTime.AsTime()}}, nil
 	case enumsspb.CALLBACK_STATE_SCHEDULED:
-		var baseURL string
 		switch v := c.Callback.GetVariant().(type) {
 		case *persistencespb.Callback_Nexus_:
+			var baseURL string
 			u, err := url.Parse(c.Callback.GetNexus().Url)
 			if err != nil {
 				return nil, fmt.Errorf("failed to parse URL: %v: %w", &c, err)
 			}
 			baseURL = u.Scheme + "://" + u.Host
+			return []hsm.Task{InvocationTask{Destination: baseURL}}, nil
+		case *persistencespb.Callback_Hsm:
+			return []hsm.Task{InvocationTask{Destination: ""}}, nil
```

Comment:
Will `Destination: ""` make the task a transfer task? or it's still an outbound task?

Reply:
https://github.com/temporalio/temporal/pull/6363/files#diff-4787d7dd87ae52614fa42a1eddf3339bf68173a4053e0a5d91062612a858d4d9R58

Reply:
That PR doesn't support immediate task without destination I believe ([link](https://github.com/temporalio/temporal/pull/6363/files#diff-4ba4ad68d17b92991c92ee717f72f03cd2efe5e7130bfac81f2c08db1f2c12abR882)). 

So the code will rely on a future change from Roey to work right?
I am ok with that, just want to confirm my understanding here.